src/FilenameParsing/FileIDHelperResolutionStrategy.php

```diff
@@ -432,10 +434,24 @@ public function findVariants($tuple, Filesystem $filesystem)
             // Make sure our yield file has an hash
             $hash = $parsedFileID->getHash() ?: $this->findHashOf($helper, $parsedFileID, $filesystem);
 
-
             // Find the correct folder to search for possible variants in
             $folder = $helper->lookForVariantIn($parsedFileID);
-            $possibleVariantsGenerator = $filesystem->listContents($folder, $helper->lookForVariantRecursive());
+
+            if (!$filesystem->directoryExists($folder)) {
+                continue;
+            }
+
+            // Use a glob if possible, as that will be more performant if the adapter supports it.
+            if (($helper instanceof GlobbableFileIDHelper) && ($filesystem instanceof GlobContentLister)) {
```

Comment:
```suggestion
            if (is_a($helper, GlobbableFileIDHelper::class) && is_a($filesystem, GlobContentLister::class)) {
```
IMO `is_a()` is preferable to `instanceof` as you don't need to be so concerned about the parans as it's now just a regular method call

Reply:
That's 1000% subjective. I prefer `instanceof`, but don't ask you to change `is_a` when I see it :p